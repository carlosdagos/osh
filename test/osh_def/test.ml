(* This module tests that we can define some programs
   as structure items *)

[%%osh_def ["cat"; "echo"]]

let%expect_test _ =
  let open Osh.Proc in
  let message = "Hello Osh!" in
  let inline_script =  echo [message] ||> cat []
  in
  run_process' inline_script;
  [%expect{|
  Hello Osh!
  |}]
