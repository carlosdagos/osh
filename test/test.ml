let () =
  let open Osh.Proc in
  run_process' begin
        [%osh "echo", ["hello world"]]
    ||> [%osh "wc", ["-l"]]
  end
