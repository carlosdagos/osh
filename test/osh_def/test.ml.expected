let cat args = Osh.Proc.create_sys_process ~program:"cat" ~args
let echo args = Osh.Proc.create_sys_process ~program:"echo" ~args
[%%expect_test
  let _ =
    let open Osh.Proc in
      let message = "Hello Osh!" in
      let inline_script = (echo [message]) ||> (cat []) in
      run_process' inline_script; [%expect {|
  Hello Osh!
  |}]]
